File: apps/merchant-web/src/schemes/order/OrderItemVariantDto.ts
export class OrderItemVariantDto {
  quantity: number;
  unitPrice: number;
  discountPercent: number;
  subTotalPrice: number;

  constructor(
    quantity: number,
    unitPrice: number,
    discountPercent: number,
    subTotalPrice: number
  ) {
    this.quantity = quantity;
    this.unitPrice = unitPrice;
    this.discountPercent = discountPercent;
    this.subTotalPrice = subTotalPrice;
  }
}

File: apps/merchant-api/PaymentService/src/PaymentService.Application/Dtos/Orders/OrderItemVariantDto.cs
namespace PaymentService.Application.Dtos.Orders;

public class OrderItemVariantDto
{
    public int Quantity { get; set; }
    public double UnitPrice { get; set; }
    public int DiscountPercent { get; set; }
    public double SubTotalPrice { get; set; }
}